app/src/main/java/com/teamwss/websoso/ui/novelRating/NovelRatingDateDialog.kt

```diff
@@ -0,0 +1,127 @@
+package com.teamwss.websoso.ui.novelRating
+
+import android.os.Bundle
+import android.view.View
+import android.widget.NumberPicker
+import androidx.fragment.app.activityViewModels
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import com.teamwss.websoso.R
+import com.teamwss.websoso.databinding.DialogNovelRatingDateBinding
+import com.teamwss.websoso.ui.common.base.BindingBottomSheetDialog
+
+class NovelRatingDateDialog :
+    BindingBottomSheetDialog<DialogNovelRatingDateBinding>(R.layout.dialog_novel_rating_date) {
+    private val viewModel: NovelRatingViewModel by activityViewModels()
+
+    override fun onViewCreated(
+        view: View,
+        savedInstanceState: Bundle?,
+    ) {
+        super.onViewCreated(view, savedInstanceState)
+        setupDataBinding()
+        setupDialogBehavior()
+        initNullDate()
+        initNumberPickerRange()
+        observeDayRange()
+        setupValueChangeListener()
+    }
+
+    private fun setupDataBinding() {
+        binding.viewModel = viewModel
+        binding.onClick = onNovelRatingButtonClick()
+        binding.lifecycleOwner = viewLifecycleOwner
+    }
+
+    private fun onNovelRatingButtonClick() =
+        object : NovelRatingClickListener {
+            override fun onDateEditClick() {}
+
+            override fun onKeywordEditClick() {}
+
+            override fun onNavigateBackClick() {}
+
+            override fun onSaveClick() {
+                viewModel.updatePreviousDate()
+                dismiss()
+            }
+
+            override fun onCancelClick() {
+                viewModel.cancelDateEdit()
+                dismiss()
+            }
+
+            override fun onClearClick() {
+                viewModel.clearCurrentDate()
+                dismiss()
+            }
+        }
+
+    private fun setupDialogBehavior() {
+        (dialog as BottomSheetDialog).behavior.state = BottomSheetBehavior.STATE_EXPANDED
+        (dialog as BottomSheetDialog).behavior.skipCollapsed = true
+    }
+
+    private fun initNullDate() {
+        viewModel.getNotNullDate()
+    }
+
+    private fun initNumberPickerRange() {
+        setupNumberPicker(binding.npRatingDateYear, 1, MAX_YEAR_VALUE, "%04d")
+        setupNumberPicker(binding.npRatingDateMonth, 1, MAX_MONTH_VALUE, "%02d")
+        setupNumberPicker(
+            binding.npRatingDateDay,
```

Comment:
a: 이거보고 띠용햇네요 똑같은함수 3번부르길래
차라리 다음처럼 고치는건 어떨까요
```suggestion
    with(binding) {
            npRatingDateDay.setupFormattedDate(MAX_YEAR_VALUE, "%04d"),
            npRatingDateYear.setupFormattedDate(),
            npRatingDateMonth.setupFormattedDate()
     }
```

Reply:
아 안그래도 지저분해서 어떻게 해야할지 고민이었는데 좋은거같네요!!